src/containers/WelcomePage/WelcomePage.jsx

```diff
@@ -162,6 +208,86 @@ export class WelcomePage extends Component {
   }
 }
 
+function mapSearchToFrontPageStructure(data, t) {
+  if (
+    !data ||
+    (data && !data.search) ||
+    (data && data.search && !data.search.results) ||
+    (data &&
+      data.search &&
+      data.search.results &&
+      data.search.results.length === 0)
+  ) {
+    return [];
+  }
+  const result = data.search.results;
+  // grouping
+  const subjects = {
+    title: `${t('contentTypes.subject-material')}:`,
+    contentType: 'results-frontpage',
+    resources: [],
+  };
+  const topics = {
+    title: `${t('subjectPage.tabs.topics')}:`,
+    contentType: 'results-frontpage',
+    resources: [],
+  };
+  const resource = {
+    title: `${t('resource.label')}:`,
+    contentType: 'results-frontpage',
+    resources: [],
+  };
+  result.map(i => {
+    if (i && i.contexts && i.contexts.length) {
+      i.contexts.map(ctx => {
+        if (!ctx.id || !ctx.resourceTypes[0]) {
+          return false;
+        }
+        const finalObj = {
+          id: `${i.id}-${ctx.id}`,
+          path: ctx.path,
```

Comment:
Path er uten rota subjects
```suggestion
           path: `/subjects${ctx.path}`,
```